src/zarr/api/asynchronous.py

```diff
@@ -393,6 +393,8 @@ async def save_array(
         _handle_zarr_version_or_format(zarr_version=zarr_version, zarr_format=zarr_format)
         or _default_zarr_version()
     )
+    if not isinstance(arr, np.ndarray):
```

Comment:
```suggestion
    if not isinstance(arr, NDArrayLike):
```

I think we support other array types (like cupy arrays). NDArrayLike is marked as `runtime_checkable` so I think that `isinstance` should work with it.